src/solvers/sat/satcheck_cadical.cpp

```diff
@@ -72,31 +89,37 @@ propt::resultt satcheck_cadicalt::do_prop_solve()
   log.statistics() << (no_variables() - 1) << " variables, " << clause_counter
                    << " clauses" << messaget::eom;
 
-  if(status == statust::UNSAT)
+  // if assumptions contains false, we need this to be UNSAT
+  for(const auto &a : assumptions)
   {
-    log.status() << "SAT checker inconsistent: instance is UNSATISFIABLE"
-                 << messaget::eom;
-  }
-  else
-  {
-    switch(solver->solve())
+    if(a.is_false())
     {
-      case 10:
-        log.status() << "SAT checker: instance is SATISFIABLE" << messaget::eom;
-        status = statust::SAT;
-        return resultt::P_SATISFIABLE;
-      case 20:
-        log.status() << "SAT checker: instance is UNSATISFIABLE"
-                     << messaget::eom;
-        break;
-      default:
-        log.status() << "SAT checker: solving returned without solution"
-                     << messaget::eom;
-        throw analysis_exceptiont(
-          "solving inside CaDiCaL SAT solver has been interrupted");
+      log.status() << "got FALSE as assumption: instance is UNSATISFIABLE"
+                   << messaget::eom;
+      status = statust::UNSAT;
+      return resultt::P_UNSATISFIABLE;
     }
   }
 
+  for(const auto &a : assumptions)
+    solver->assume(a.dimacs());
```

Comment:
Why do we need to do this now? This wasn't here previously?

Reply:
Support for assumptions simply was missing before.